sdk/core/System.ClientModel/tests/TestFramework/Mocks/MockPipelineResponseTransport.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace ClientModel.Tests.Mocks;
+
+public class MockPipelineResponseTransport : PipelineTransport
```

Comment:
Looks like we're adding a third mock PipelineTransport?  Would it be possible to use either MockPipelineTransport or ObservableTransport for this instead of adding a new one?

I made some updates to [MockPipelineTransport here](https://github.com/annelo-msft/openai-in-typespec/blob/azoai-client-inheritance2/tsp-output/%40azure-tools/typespec-csharp/app/DemoApp/Mocks/MockPipelineTransport.cs), but haven't rolled them back into ClientModel yet.  I can do that in a separate PR if that would be helpful, or if you wanted to include them in this PR, you could do that too?

Reply:
Yes I think I should be able to add them to MockPipelineTransport. When I tried doing that initially I just kept hitting tons of NotImplementedExceptions at every step and it felt like I was writing a lot more code than what I needed with this approach - but it's probably worthwhile to condense these types into the existing mocks.